### PR TITLE
[GVN] Skip debug instructions in findDominatingValue function

### PR DESCRIPTION
**Pull request by skachkov-llvm**
findDominatingValue has a search limit, and when it is reached, optimization is
not applied. This patch fixes the issue that this limit also takes into account
debug intrinsics, so the result of optimization can depend from the presence of
debug info.

### CODE REVIEW

**Comment by skachkov-sc** [comment]
Note: https://reviews.llvm.org/D141680#4643142

File: llvm/test/Transforms/GVN/load-through-select-dbg.ll [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+!1 = !DIFile(filename: "bbi-78272.c", directory: "/tmp")
+!5 = !DIBasicType(name: "int", size: 16, encoding: DW_ATE_signed)
+
+!35 = !{i32 7, !"Dwarf Version", i32 4}
+!36 = !{i32 2, !"Debug Info Version", i32 3}
+!40 = !DILocation(line: 15, column: 7, scope: !41)
+!41 = distinct !DISubprogram(name: "x", scope: !1, file: !1, line: 14, type: !42, scopeLine: 14, flags: DIFlagAllCallsDescribed, spFlags: DISPFlagLocalToUnit | DISPFlagDefinition | DISPFlagOptimized, unit: !0, retainedNodes: !45)
+!42 = !DISubroutineType(types: !43)
+!43 = !{!5}
+!45 = !{!46, !47, !48, !49, !50}
```
Comment (mikaelholmen) [inline llvm/test/Transforms/GVN/load-through-select-dbg.ll:41]: I think you can remove !47, !48, !49 and !50

**Comment by mikaelholmen** [comment]
I've verified that this solves the problem I reported. Thanks!

File: llvm/test/Transforms/GVN/load-through-select-dbg.ll (hunk 1 above)
Comment (skachkov-sc) [reply llvm/test/Transforms/GVN/load-through-select-dbg.ll]: Done

**Review by mikaelholmen — APPROVED** [review]
Thanks!

**Pull request merged**